### PR TITLE
Port optimizations for primitive set operations to SetIterables.

### PR DESCRIPTION
**Pull request by donraab**
Signed-off-by: Donald Raab <Donald.Raab@bnymellon.com>

### CODE REVIEW

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/internal/SetIterables.java [hunk 1]
```diff
@@ -51,8 +52,16 @@ private SetIterables()
             SetIterable<? extends E> setB,
             R targetSet)
     {
-        Iterate.addAllIterable(setA, targetSet);
-        Iterate.addAllIterable(setB, targetSet);
+        if (setA.size() > setB.size())
```
Comment (mohrezaei) [inline eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/internal/SetIterables.java:55]: the order doesn't seem to make any difference here...

**Review by mohrezaei — CHANGES_REQUESTED** [review]

**Comment by mohrezaei** [comment]
For optimization code, it's good to add an acceptance test.

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/internal/SetIterables.java (hunk 1 above)
Comment (donraab) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/internal/SetIterables.java:55]: Yes, I changed back to optimizing `union` so I can use `UnifiedSet.newSet()`, and reverted `unionInto`.

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/internal/SetIterables.java [hunk 2]
```diff
@@ -43,7 +42,13 @@ private SetIterables()
             SetIterable<? extends E> setA,
             SetIterable<? extends E> setB)
     {
-        return SetIterables.unionInto(setA, setB, UnifiedSet.newSet());
+        if (setA.size() > setB.size())
+        {
+            UnifiedSet<E> result = UnifiedSet.newSet(setA);
```
Comment (mohrezaei) [inline eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/internal/SetIterables.java:47]: The following code should reduce bytecode size and increase inlining chance:
```java
if (setA.size() > setB.size())
{
    SetIterable<? extends E> tmp = setA;
    setA = setB;
    setB = tmp;
}
```
(and below)

**Review by mohrezaei — CHANGES_REQUESTED** [review]

**Comment by prathasirisha** [comment]
Hi @donraab, 

1. Noticed that the javadoc for `intersect` is a bit inaccurate. With the changes you are making, this line is not valid - `The output will contain instances from {@code this}, not {@code set}.`

2. copyright information needs to be updated.

**Comment by donraab** [comment]
Hi @prathasirisha, I wil fix the Javadoc in `SetIterable` in a separate PR. I'm only focusing changes in the `SetIterables` utility in this PR. I'm adding a test for the class as well since there was no dedicated test class. I've ported the tests you added for the primitive union, intersect and difference, and added a test for validating the performance improvement of `union` over `unionInto`.

**Review by mohrezaei — APPROVED** [review]

**Comment by donraab** [comment]
The TeamCity inspections 2 build is failing for a seemingly unrelated change.

**Pull request merged**